apps/dashboard/src/components/primitives/editor.tsx

```diff
@@ -159,27 +169,27 @@ export const Editor = React.forwardRef<{ focus: () => void; blur: () => void },
       [fontFamily]
     );
 
-    const debouncedOnChange = useMemo(
-      () =>
-        debounce((value: string) => {
-          if (onChange) onChange(value);
-        }, 50),
+    const onChangeCallback = useCallback(
+      (value: string) => {
+        // when typing fast the onChange event is called multiple times during one render phase
+        // by default react batches state updates and only triggers one render phase
+        // which results in value not being updated and "jumping" effect in the editor
+        // to prevent this we need to flush the state updates synchronously
+        flushSync(() => {
+          onChange?.(value);
+        });
+      },
```

Comment:
The "lagging" in the editor was caused by two problems:
- RHF `useWatch` causing re-render for many form components
- React batching state updates

This code solves the second.